getting_started/scripting/c_sharp/c_sharp_style_guide.rst

```diff
@@ -0,0 +1,298 @@
+.. _doc_c_sharp:
+
+Style Guide
+===========
+
+Having well-defined and consistent coding conventions is important for every project, and Godot
+is no exception to this rule.
+
+This page contains a coding style guide which is followed by developers and contributors of Godot
+itself. As such, it is mainly intended for those who want to contribute to the project, but since
+the conventions and guidelines mentioned in this article are those most widely adopted by the users
+of the language, we encourage you to do the same, especially if you do not have such a guide yet.
+
+.. note:: This article is by no means an exhaustive guide on how to follow the standard coding
+        conventions or best practices. If you feel unsure of an aspect which is not covered here,
+        please refer to more comprehensive documentation, such as
+        `C# Coding Conventions <https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions>`_ or
+        `Framework Design Guidelines <https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines>`_.
+
+Language Specification
+----------------------
+
+Currently, Godot uses C# version 6.0 in its engine and example source code. So, before we move to
+a newer version, care must be taken to avoid mixing language features only available in C# 7.0 or
+later, such as pattern matching or expression-bodied members inside get/set accessors.
+
+For detailed information of C# features in different versions, please see
+`What's New in C# <https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/>`_.
+
+Formatting Conventions
+----------------------
+
+* If you create a new file, make sure that it uses linefeed(*LF*) characters to break lines, not *CRLF* or *CR*.
+* Use UTF-8 encoding, if your code contains non-ASCII characters,
```

Comment:
UTF-8 *without byte order mark (BOM)*, *always* (drop the non-ASCII characters condition, IMO)